core/scale/type_decoder.hpp

```diff
@@ -64,6 +64,22 @@ namespace kagome::scale {
       return boolean::decodeTribool(stream);
     }
   };
+
+  /**
+   * @class TypeDecoder<std::pair<F,S>> is specialization of TypeDecoder class
+   * it implements decoding std::pair from stream
+   * @tparam F first type
+   * @tparam S second type
+   */
+  template <class F, class S>
+  struct TypeDecoder<std::pair<F, S>> {
+    using value_type = std::pair<F, S>;
+    outcome::result<value_type> decode(common::ByteStream &stream) {
+      OUTCOME_TRY(first_value, TypeDecoder<F>{}.decode(stream));
```

Comment:
Please leave todo with task to refactor it to free function

Reply:
Done